jStorage.github.js

```diff
@@ -58,49 +33,97 @@
             }
         };
 
-
-        //xmlhttp.setRequestHeader('Accept', 'application/vnd.github.raw+json');
         xmlhttp.setRequestHeader('Accept', 'application/json;charset=UTF-8');
         xmlhttp.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
-        //xmlhttp.setRequestHeader("Access-Control-Allow-Origin", window.location.protocol + '//' + window.location.host);
-        //xmlhttp.setRequestHeader('Content-Type', 'application/xml');
         if (token) {
             xmlhttp.setRequestHeader('Authorization', 'token ' + token);
         }
 
         data ? xmlhttp.send(JSON.stringify(data)) : xmlhttp.send();
     }
 
-
-
     jStorage.providers.github = {
         init: function (wrapper, config) {
             var self = this;
 
-            console.log('github init');
-
             this._config = config;
             this._hasCallback = config && typeof (config.callback) === "function";
             this._state = 'random' + new Date().getTime();
             this._code = false;
             this._shaCache = {};
 
-            if (this.ensureAuth(wrapper, config)) {
-                // we have valid auth, call callback...
-                if (self._hasCallback) {
-                    try {
-                        self._config.callback(wrapper, callStatus);
-                    } catch (e) {
-                        // TODO: handle error
-                    }
-                }
+            this.ensureAuth(wrapper, config);
+
+            // https://github.com/settings/tokens/new
+            var token = this._config.token;
```

Comment:
all github related in this pr has been fixed in a refactoring.
Can be viewed at #12